finetuner-workflow/finetune-workflow.yaml

```diff
@@ -0,0 +1,300 @@
+        value: "{{inputs.parameters.wandb_key}}"
+      - name: PYTHONUNBUFFERED
+        value: "1"
+      - name: TORCH_EXTENSIONS_DIR
+        value: "{{inputs.parameters.torch_cache}}"
+      resources:
+        requests:
+          memory: 128Gi
+          cpu: "8"
+        limits:
+          nvidia.com/gpu: 1
+      volumeMounts:
+        - mountPath: "/{{workflow.parameters.pvc}}"
+          name: "{{workflow.parameters.pvc}}"
+    volumes:
+      - name: "{{workflow.parameters.pvc}}"
+        persistentVolumeClaim:
+           claimName: "{{workflow.parameters.pvc}}"
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: gpu.nvidia.com/vram
+              operator: In
+              values:
+              - "48"
+            - key: topology.kubernetes.io/region
+              operator: In
+              values:
+              - "{{workflow.parameters.region}}"
+
+  - name: model-inference-service
+    inputs:
+      parameters:
+        - name: model_path
+    resource:
+      action: apply
+      manifest: |
+        apiVersion: serving.kubeflow.org/v1beta1
+        kind: InferenceService
+        metadata:
+          name: inference-{{ workflow.parameters.run_name }}
+          annotations:
+            autoscaling.knative.dev/scaleToZeroPodRetentionPeriod: 20m
+        spec:
+          predictor:
+            minReplicas: 0
+            maxReplicas: 1
+            affinity:
+              nodeAffinity:
+                requiredDuringSchedulingIgnoredDuringExecution:
+                  nodeSelectorTerms:
+                  - matchExpressions:
+                      - key: gpu.nvidia.com/vram
+                        operator: In
+                        values:
+                        - "24"
+                preferredDuringSchedulingIgnoredDuringExecution:
+                - weight: 10
+                  preference:
+                    matchExpressions:
+                      - key: topology.kubernetes.io/region
+                        operator: In
+                        values:
+                          - {{ workflow.parameters.region }}
+            containers:
+              - name: kfserving-container
+                image: {{ workflow.parameters.inference_image }}
+                imagePullPolicy: IfNotPresent
+                command:
+                  - "python3"
+                  - "/inference/huggingface.py"
+                env:
+                  - name: MODEL_NAME
+                    value: "final"
+                  - name: MODEL_PRECISION
+                    value: "fp16"
+                  - name: STORAGE_URI # KFServing mounts the PVC at /mnt/pvc/
+                    value: pvc://{{ workflow.parameters.pvc }}/
+                  - name: MODEL_PATH
+                    value: /mnt/pvc/{{ inputs.parameters.model_path }}
+                resources:
+                  requests:
```

Comment:
Let's make requests match limits here.

Reply:
Done, except for the trainer which has limits set to 192gb.